data-protocols/ids/ids-api-configuration/src/main/java/org/eclipse/dataspaceconnector/ids/api/configuration/IdsApiConfigurationExtension.java

```diff
@@ -61,8 +65,12 @@ public void initialize(ServiceExtensionContext context) {
         }
         
         monitor.info(format("IDS API will be available at [path=%s], [port=%s].", path, port));
-        
-        context.registerService(IdsApiConfiguration.class, new IdsApiConfiguration(contextAlias, path));
+
+        var idsWebhookAddress = context.getSetting(IDS_WEBHOOK_ADDRESS, DEFAULT_IDS_WEBHOOK_ADDRESS);
+        var webhookPath = path + (path.endsWith("/") ? "data" : "/data");
+        idsWebhookAddress = idsWebhookAddress + webhookPath;
```

Comment:
NB: this code was moved from `IdsMultipartDispatcherServiceExtension`

Reply:
nit: while moving this code you could refactor it using the `java.net.URL` class for the path operations.

Reply:
That class being "very old" is not very developer friendly, an alternative could be to use OkHttp's URLBuilder class as well